boa/src/builtins/number/mod.rs

```diff
@@ -159,6 +159,145 @@ pub fn to_precision(this: &mut Value, args: &[Value], _ctx: &mut Interpreter) ->
+    // either way, with additional space for sign, decimal point and string
+    // termination should be sufficient.
+    let mut buffer: [u8; BUF_SIZE] = [0; BUF_SIZE];
+    let mut integer_cursor = BUF_SIZE / 2;
+    let mut fraction_cursor = integer_cursor;
+    // let negative: bool = value < 0f64;
+    let negative = value.is_sign_negative();
+    if negative {
+        value = -value
+    }
+    // Split the value into an integer part and a fractional part.
+    let mut integer = value.trunc();
+    let mut fraction = value.fract();
+
+    // We only compute fractional digits up to the input double's precision.
+    let mut delta = 0.5 * (next_after(value, f64::MAX) - value);
+    delta = next_after(0.0, f64::MAX).max(delta);
+    assert!(delta > 0.0);
+    if fraction >= delta {
+        // Insert decimal point.
+        let fresh0 = fraction_cursor;
+        fraction_cursor += 1;
+        buffer[fresh0] = b'.';
+        loop {
+            // Shift up by one digit.
+            fraction *= radix as f64;
+            delta *= radix as f64;
+            // Write digit.
+            let digit = fraction as u32;
+            buffer[fraction_cursor] = std::char::from_digit(digit, radix).unwrap() as u8;
+            fraction_cursor += 1;
+            // Calculate remainder.
+            fraction -= digit as f64;
+            // Round to even.
+            if fraction > 0.5
+                || (fraction - 0.5).abs() < f64::EPSILON && digit & 1 != 0 && fraction + delta > 1.0
+            {
+                loop {
+                    // We need to back trace already written digits in case of carry-over.
+                    fraction_cursor -= 1;
+                    if fraction_cursor == BUF_SIZE / 2 {
+                        //              CHECK_EQ('.', buffer[fraction_cursor]);
+                        // Carry over to the integer part.
+                        integer += 1.;
+                        break;
+                    } else {
+                        let c: u8 = buffer[fraction_cursor];
+                        // Reconstruct digit.
+                        let digit_0 = (c as char).to_digit(10).unwrap();
+                        if digit_0 + 1 >= radix {
+                            continue;
+                        }
+                        buffer[fraction_cursor] =
+                            std::char::from_digit(digit_0 + 1, radix).unwrap() as u8;
+                        fraction_cursor += 1;
+                        break;
+                    }
+                }
+                break;
+            }
+            if fraction < delta {
+                break;
+            }
+        }
+    }
+    // Compute integer digits. Fill unrepresented digits with zero.
+    //  while (Double(integer / radix).Exponent() > 0) {
+    // let mut exp = 0;
+    // unsafe {
+    //     frexp(integer / radix as f64, &mut exp);
+    // }
+
+    while frexp(integer / radix as f64).1 > 0 {
```

Comment:
Try to use `f64::from(radix)` instead of `radix as f64`, this will make it easier to track errors in the future, and prevents the compilation of invalid conversions. There are multiple occurrences of this.
```suggestion
    while frexp(integer / f64::from(radix)).1 > 0 {
```

One extra option is to generate the `f64::from(radix)` once at the beginning, and re-use it. This should be more performant.